### PR TITLE
adding region list- ordered and changed preferred region map to handl…

### PR DESCRIPTION
**Pull request by VaclavK**
@djfr  this is what we talked about yesterday  I believe

### CODE REVIEW

**Comment by VaclavK** [comment]
should we change Regions from strings to Enum?

**Comment by djfr** [comment]
@VaclavK I think we should at this point, but we still need to provide some sort of an API to give those constants, because that's what the CosmosDB client uses as preferred regions.

**Comment by VaclavK** [comment]
@djfr I added the enum but kept the deployment context as it was (pure strings) - partially as the interface is coming from core nuget

**Comment by VaclavK** [comment]
@djfr refactored the enum extension to capture the values via dedicated attribute as we talked about

File: src/Eshopworld.DevOps/RegionsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        /// <returns>short string value - code - of the region</returns>
+        public static string ToRegionCode(this Regions it)
+        {         
+            return GetAttributeInstance(it).ToShortString();
+        }
+
+        private static RegionDescriptorAttribute GetAttributeInstance(Regions it)
+        {
+            FieldInfo fi = it.GetType().GetField(it.ToString());
+            return (RegionDescriptorAttribute)fi.GetCustomAttributes(
```
Comment (arturek) [inline src/Eshopworld.DevOps/RegionsExtensions.cs]: possible NRE when the value of 'it' is invalid.

Comment (VaclavK) [reply src/Eshopworld.DevOps/RegionsExtensions.cs]: thanks, spot on

**Pull request merged**